libpromises/cf3parse_logic.h

```diff
@@ -1028,4 +1031,63 @@ static inline void ParserBeginBundleBody()
     P.useargs = NULL;
 }
 
+static inline void ParserHandleQuotedIlistItems()
+{
+    if (StringContainsUnresolved(P.currentstring))
+    {
+        RlistAppendScalar((Rlist **) &P.currentRlist, P.currentstring);
+        FREE_AND_NULL(P.currentstring);
+        return;
+    }
+
+    double value;
+    bool ok = DoubleFromString(P.currentstring, &value);
+    if (!ok)
+    {
+        // TODO: we dont really want DoubleFromStrings error output...
+        ParseError("Lval '%s' expects numbers", P.lval);
+        FREE_AND_NULL(P.currentstring);
+        return;
+    }
+
+    if (StringContainsChar(P.currentstring, '.'))
+    {
+        ParseWarning(PARSER_WARNING_TRUNCATED,
+                     "Real number '%s' will be truncated",
+                     P.currentstring);
+    }
+
+    char ret[PRINTSIZE(intmax_t)] = {0};
+    xsnprintf(ret, PRINTSIZE(intmax_t), "%jd", (intmax_t)value);
+
+    RlistAppendScalar((Rlist **) &P.currentRlist, ret);
+    FREE_AND_NULL(P.currentstring);
+}
+
+static inline void ParserHandleQuotedRlistItems()
+{
+    if (StringContainsUnresolved(P.currentstring))
+    {
+        RlistAppendScalar((Rlist **) &P.currentRlist, P.currentstring);
+        FREE_AND_NULL(P.currentstring);
+        return;
+    }
+
+    double value;
+    bool ok = DoubleFromString(P.currentstring, &value);
+    if (!ok)
+    {
+        // TODO: we dont really want DoubleFromStrings error output...
+        ParseError("Lval '%s' expects numbers", P.lval);
+        FREE_AND_NULL(P.currentstring);
+        return;
+    }
+
+    char ret[PRINTSIZE(double)] = {0};
+    xsnprintf(ret, PRINTSIZE(double), "%lf", value);
```

Comment:
I don't think `PRINTSIZE()` works for `double`.

Reply:
```c
#include <stdio.h>

#define PRINTSIZE(what) (3 + 53 * sizeof(what) / 22)

int main() {
    printf("%d", PRINTSIZE(double));
}
```
Locally the above program works and outputs `22`

Reply:
And how long is the biggest double value with the highest possible precision?

Reply:
i looked more closely into printsize and indeed it only works for integrals and not IEEE floating point dark magic. However that leaves me in a predicament.

I gave a ludicrously large number as list item and i crossreferenced that with
`#define CF_REALRANGE "-9.99999E100,9.99999E100"` which gives 101 digits (excluding dot, sign, decimals and null terminator)
Decimals seem to not influence the parsing but im still investigating (on the other hand we seem to be interested in 6 decimal precision so that may not influence the problem too much).

With that in mind does it make sense to allocate a buffer of:
```
sign   digits   dot   "forced" precision   '\0'
1    + 101    + 1   + 6                  + 1    = 120 characters?
```
And use snprintf:
```c
#define DOUBLE_PRINTSIZE (1 + 101 +  1 + 6 + 1)
char ret[DOUBLE_PRINTSIZE] = {0};

snprintf(ret, DOUBLE_PRINTSIZE, "%.6lf", value);
```

Reply:
What if people actually care about the precision? Floating point numbers are designed in a way to support storage of small numbers with high precision or large numbers with low precision. We should support both. The easiest case might be just using the string from the policy without the conversion there and back in cases where that is possible.